src/sidebar/store/use-store.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+/* global process */
+
+const shallowEqual = require('shallowequal');
+const { useEffect, useRef, useReducer } = require('preact/hooks');
+
+const { useService } = require('../util/service-context');
+const warnOnce = require('../util/warn-once');
+
+/**
+ * Hook for accessing state or actions from the store inside a component.
+ *
+ * This hook fetches the store using `useService` and returns the result of
+ * passing it to the provided callback. The callback will be re-run whenever
+ * the store updates and the component will be re-rendered if the result of
+ * `callback(store)` changed.
+ *
+ * This ensures that the component updates when relevant store state changes.
+ *
+ * @example
+ *   function MyWidget({ widgetId }) {
```

Comment:
I'm having trouble following this example. I have a feeling it may in part be because of the use of the `useCallback` hook.

Reply:
Thanks for the feedback. The `useCallback` dates from an earlier version of the code. It isn't required so I removed it.